### PR TITLE
ARC-838 builds doc

### PR DESCRIPTION
**Pull request by rachellerathbone**
Doc to provide example of builds.yml file and to explain how to see builds data show up in multiple issue keys when more than one commit is involved.

For easier viewing: https://github.com/atlassian/github-for-jira/blob/ARC-838-builds-doc/docs/builds.md

Related to changes for https://github.com/atlassian/github-for-jira/pull/951

### CODE REVIEW

**Review by yvonnevictoria — APPROVED** [review]

File: docs/builds.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+we need a base branch and a head branch. When the app has both of these, we are able to make a request to GitHub to compare your
+commits. On the flipside, if you only listed `main`, for instance, there would be no way for the app to ask GitHub for a comparison.
+Instead, the best it can do is use the data GitHub sends in the response when the `workflow_run` event is triggered,
+which only includes the most recent commit. This means that if a developer were to make multiple commits, perhaps on multiple branches, and
+reference various Jira issue keys in each commit, GitHub would only send our app data about the latest commit. In turn, this
+would mean that we could only extract any issue keys from that single message. Although there may be numerous Jira issues
+involved, in this scenario, you would only see builds data for any issue keys from the latest commit message.
+
+When you list branches under `pull_request` you'll need to be very specific about the branches you want our app to target. If any branch is created that isn't listed here, the app won't be able to compare your commits and send the most accurrate data to Jira.
```
Comment (yvonnevictoria) [inline docs/builds.md:77]: mistype - `accurate`

File: docs/builds.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+One important thing to note in the above example is the branches being targeted under `pull_requests` `branches`:
+
+```
+pull_request:
+    branches:
+      - main
+      - feature/**
+```
+
+In order for the GitHub for Jira app to be able to compare two points in your git history (possible once a pull request is opened)
```
Comment (yvonnevictoria) [inline docs/builds.md:68]: `In order for the Github for Jira app to compare` is a bit easier to read imo - nit

I would also remove the brackets and turn them into a new sentence to say 
``` 
In order for the GitHub for Jira app to be able to compare two points in your git history, we need a base branch and a head branch. This is only possible once a pull request is opened. When the app has both branches...
```
I'm not sure if this is what you were trying to say in the brackets but I felt it was unclear

Comment (rachellerathbone) [reply docs/builds.md:68]: Screen Shot 2021-12-07 at 12 37 26 pm
 
Just ran the suggestion through a grammar check and it's highlights that the additional words make the sentence too wordy. Will leave this as is.

Comment (rachellerathbone) [reply docs/builds.md:68]: Scratch that. Ended up rewording that first part to make the flow between sentences make more sense.

File: docs/builds.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+      # Runs a set of commands using the runners shell
+      - name: Run a multi-line script
+        run: |
+          echo Add other actions to build,
+          echo test, and deploy your project.
+          sleep 60s
+```
+
+Once you have a similar file in any of your repositories that are connected to our app, you will start to see builds data
```
Comment (bgvozdev) [inline docs/builds.md:52]: nit: maybe `to our app` -> `to your Jira via the app`

File: docs/builds.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    branches:
+      - main
+      - feature/**
+```
+
+When you open a pull request in a connected repository, the GitHub for Jira app can compare two points in your git history is we have access to a base branch and a head branch. If the app has both of these, it can make a request to GitHub to compare your
+commits. On the flipside, if you only listed `main`, for instance, there would be no way for the app to ask GitHub for a comparison.
+Instead, the best it can do is use the data GitHub sends in the response when the `workflow_run` event is triggered,
+which only includes the most recent commit. This means that if a developer were to make multiple commits, perhaps on multiple branches, and
+reference various Jira issue keys in each commit, GitHub would only send our app data about the latest commit. In turn, this
```
Comment (bgvozdev) [inline docs/builds.md:72]: nit: maybe `only send our app data` -> `only send the data to Jira`

File: docs/builds.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+would mean that we could only extract any issue keys from that single message. Although there may be numerous Jira issues
+involved, in this scenario, you would only see builds data for any issue keys from the latest commit message.
+
+When you list branches under `pull_request` you'll need to be very specific about the branches you want our app to target. If any branch is created that isn't listed here, the app won't be able to compare your commits and send the most accurate data to Jira.
```
Comment (bgvozdev) [inline docs/builds.md:76]: nit maybe `you want our app to target` -> `you want the app to target`?

**Review by bgvozdev — APPROVED** [review]

**Review by bgvozdev — APPROVED** [review]

**Review by yvonnevictoria — APPROVED** [review]
